### PR TITLE
👌 IMPROVE: move timeline to css grid layout and restructure code

### PR DESCRIPTION
**Pull request by violoncelloCH**
* Target version: master 

### Summary
this is mainly a restructuring of the code and switches the timeline to a css grid layout (thanks for the help @skjnldsv !) it gives us a nicer alignment of the elements in the timeline.

for normal posts (no boost or notification) it's not yet working correctly as the first row which is empty in this case still takes up it's space (it shouldn't)... @skjnldsv any idea how to fix that?

### CODE REVIEW

File: src/components/TimelineEntry.vue [hunk 1]
```diff
@@ -102,10 +112,57 @@ export default {
...
 <style scoped lang="scss">
 	.timeline-entry {
+		display: grid;
+	    grid-template-columns: 44px 1fr;
+	    grid-template-rows: 30px 1fr;
 		padding: 10px;
 		margin-bottom: 10px;
 	}
 
+<<<<<<< Updated upstream
```
Comment (skjnldsv) [inline src/components/TimelineEntry.vue]: rebase issue

**Comment by icewind1991** [comment]
rebased and fixed the layout issue with normal posts

Before:
image

After:
image

**Comment by daita** [comment]
please rebase!

**Comment by icewind1991** [comment]
rebased

**Comment by daita** [comment]
I encounter few issues: 
- Actor URL with trailing slash are not working anymore: http://test.local/index.php/apps/social/@ncdev/
- JS error on timeline:

```
[Vue warn]: Property or method "getSinglePostTimeline" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property. See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.

found in

---> <TimelineEntry> at src/components/TimelineEntry.vue
       <TransitionGroup>
         <Timeline> at src/components/TimelineList.vue
           <Timeline> at src/views/Timeline.vue
             <AppContent>
               <Content>
                 <App> at src/App.vue
                   <Root> vue.esm.js:628


[Vue warn]: Invalid handler for event "click": got undefined

found in

---> <TimelineEntry> at src/components/TimelineEntry.vue
       <TransitionGroup>
         <Timeline> at src/components/TimelineList.vue
           <Timeline> at src/views/Timeline.vue
             <AppContent>
               <Content>
                 <App> at src/App.vue
                   <Root> vue.esm.js:628


```

**Comment by icewind1991** [comment]
both should be fixed now

**Pull request merged**